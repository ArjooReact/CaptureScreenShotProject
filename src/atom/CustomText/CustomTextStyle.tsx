import {StyleSheet} from 'react-native';
import Dimension from '../../utility/Dimension';
import Colors from '../../utility/Colors';
const styles = StyleSheet.create({
  fontStyle: {
    fontSize: Dimension.FontSize,
    color: Colors.WHITE,
  },
});
export default styles;
