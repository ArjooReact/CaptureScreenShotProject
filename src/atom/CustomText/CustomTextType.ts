 export interface CustomTextType{
    textTitle:string
 }