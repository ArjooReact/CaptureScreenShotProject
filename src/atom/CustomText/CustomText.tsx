import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {CustomTextType} from './CustomTextType';
import styles from './CustomTextStyle';
const CustomText: React.FC<CustomTextType> = ({textTitle}) => {
  return <Text style={styles.fontStyle}>{textTitle}</Text>;
};
export default CustomText;
