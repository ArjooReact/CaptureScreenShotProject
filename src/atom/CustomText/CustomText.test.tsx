import renderer from 'react-test-renderer';
import {test, jest, describe} from '@jest/globals';
import {expect, it, beforeEach} from '@jest/globals';
import CustomText from './CustomText';


describe('all mock', () => {
 
  let value: number = 1;
  let htmlWrapper: any;
 
  test('testcase for matching Snapshot==>',()=>{
    const AppRef = renderer.create(<CustomText textTitle="Submit" />);
    expect(AppRef.toJSON()).toMatchSnapshot();
  })

  test('testcase for checking dom element for CustomButton==>', () => {
    const AppRef = renderer.create(<CustomText textTitle="Submit" />);
    //console.log('arzoo test======', AppRef.toJSON());
    let props: any = AppRef.toJSON();
    //console.log('Props.....', props?.children[0]);
    expect(AppRef.toJSON()).toBeTruthy();
  });

  test('testcase for checking child element for CustomButton==>',()=>{
    const AppRef = renderer.create(<CustomText textTitle="Submit" />);
    let props: any = AppRef.toJSON();
    expect( props?.children[0]).toBe('Submit')
  }) 
});
