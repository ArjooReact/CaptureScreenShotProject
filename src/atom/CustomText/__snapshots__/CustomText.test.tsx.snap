// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all mock testcase for matching Snapshot==> 1`] = `
<Text
  style={
    {
      "color": "#ffffff",
      "fontSize": 16,
    }
  }
>
  Submit
</Text>
`;
